src/js_realm.hpp

```diff
@@ -1285,5 +1292,137 @@ void RealmClass<T>::privileges(ContextType ctx, ObjectType this_object, Argument
 #endif
 }
 
+class InvalidNameException : public std::invalid_argument {
+    public:
+        InvalidNameException(const std::string& __s) : std::invalid_argument(__s) {}
```

Comment:
`__s` isn't a legal variable name (it's reserved for the implementation and using it is UB).

Reply:
There also doesn't appear to be any reason for this type to exist at all; there's no benefit to throwing a custom subclass of `invalid_argument` rather than just throwing `invalid_argument` unless there's something that'd want to catch that specific subclass, which will never be the case when the calling code is JS.

Reply:
I'll remove that - no need for it now that I've implemented a simpler solution.